nameko/concurrency/__init__.py

```diff
@@ -0,0 +1,224 @@
+"""Concurrency abstractions.
+
+Allows concurrency primitives to be abstracted such that both eventlet and
+gevent can be used to run nameko. The backend is chosen at import time and may
+be set using the global nameko config object. Valid values are "eventlet" and
+"gevent".
+
+To write nameko code and extensions that is compatible with multiple
+concurrency backend, use this compatability layer package instead of importing
+from a specific backend:
+
+NO:
+
+>>> from eventlet import sleep
+
+YES:
+
+>>> from nameko.concurrency import sleep
+
+"""
+import os
+import sys
+
+from nameko import config
+from nameko.constants import (
+    CONCURRENCY_BACKEND_CONFIG_KEY, DEFAULT_CONCURRENCY_BACKEND,
+    MONKEY_PATCH_ENABLED_CONFIG_KEY, DEFAULT_MONKEY_PATCH_ENABLED
+)
+
+mode = os.environ.get(
+    CONCURRENCY_BACKEND_CONFIG_KEY,
+    config.get(CONCURRENCY_BACKEND_CONFIG_KEY, DEFAULT_CONCURRENCY_BACKEND)
```

Comment:
config is always empty, because the module is always before load file, so is not work fine in config.yaml